guides/no-monadfail-st-inst.md

```diff
@@ -0,0 +1,71 @@
+# What
+
+CLC has approved the [proposal to remove the `ST` instance of the `MonadFail` class 
+from `base`][proposal]. `MonadFail` consists of the single method `fail`, which is
+meant to be implemented in terms of the monad itself. In contrast, the `ST` instance
+currently (GHC 9.2, `base-4.16.0.0`) has:
+
+```haskell
+    fail s = errorWithoutStackTrace s
+```
+
+This goes against the above-mentioned idea behind the `fail` method. 
+
+[proposal]: https://github.com/haskell/core-libraries-committee/issues/33
+
+# When
+
+At the very earliest the change may appear in GHC 9.4 (~ Q3 2022), but
+given that a feature freeze for 9.4 is coming soon, it's more likely to
+happen in GHC 9.6 (~ Q2 2023).
```

Comment:
```suggestion
At the very earliest the change may appear in GHC 9.4 (~ Q3 2022).
```
Given that there are no prerequisites, I'd expect this to happen in 9.4.

Reply:
Thanks!